### DISCUSSION TITLE
Can I add a new named vector in an exist collection?

### DISCUSSION THREAD

**Discussion by xtyDoge** [qdrant/qdrant#2543] [Q&A] [answered]
I want to split my collection, and shard by month. I try to use named vector to do this. That's the steps I do

1. Create a collection with named vector, with a named vector `202308`
```
PUT /collections/t_time_shard_collection
{
    "vectors":{
        "202308":{
            "size":4,
            "distance":"Dot",
            "hnsw_config":{
                "m":16,
                "ef_construct":128,
                "full_scan_threshold":10,
                "max_indexing_threads":3,
                "on_disk":false
            }
        }
    },
    "shard_number":2
}
```
2. Then I try to update this collection to add a new named vector called `202309`
```
PATCH /collections/t_time_shard_collection
{
    "vectors":{
        "202309":{
            "size":4,
            "distance":"Dot",
            "hnsw_config":{
                "m":16,
                "ef_construct":128,
                "full_scan_threshold":10,
                "max_indexing_threads":3,
                "on_disk":false
            }
        }
    }
}
```
 But it fails and I get the response
```
{
  "error": "Wrong input: Not existing vector name error: 202309"
}
```

So how can I add named vector to exsited collection? I want to split the whole collection to shards because I want to use different params on shards. For example,  vector and hnsw indexes of data in  recently 3 month should be loaded in RAM, and others should be stored on disk to save memory. Can I do it in another way if shard by named vector is not a solution.

**Comment by timvisee** [comment, marked as answer]
At this time, it is not possible to add a new named vector to an existing collection. Nor is it possible to shard dynamically, but who knows what t he future might hold.

What about this approach: partitioning the data over two collections? Both collections would have just one vector type. One is for recent/hot data, the other is for old/cold data. You configure the hot collection to always live in RAM, and configure the cold collection to always live on disk.  You may then set up some form of cron job to transfer _old_ points from hot to cold on a daily basis. You can keep track of the age of all points by adding payload having a timestamp.

This is of course a very manual approach, but it at least gives you control over the situation. You'll easily be able to tune both collections and the period of points you transfer.

Maybe this documentation page can give you another insight as well: https://qdrant.tech/documentation/tutorials/multiple-partitions/
